pkg/apis/v1alpha2/aksnodeclass.go

```diff
@@ -27,6 +27,11 @@ import (
 // AKSNodeClassSpec is the top level specification for the AKS Karpenter Provider.
 // This will contain configuration necessary to launch instances in AKS.
 type AKSNodeClassSpec struct {
+	// VnetSubnetID is the subnet used by nics provisioned with this nodeclass.
+	// If not specified, we will use the default --vnet-subnet-id specified on the managed cluster
```

Comment:
todo: Add a aka.ms/aks/nap/subnet-id tsg for making sure they specify a correct subnet without overlapping cidr etc.